lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -207,7 +207,9 @@ def _path_to_3d_segment(path, zs=0, zdir='z'):
 def _paths_to_3d_segments(paths, zs=0, zdir='z'):
     """Convert paths from a collection object to 3D segments."""
 
-    zs = np.broadcast_to(zs, len(paths))
+    if not np.iterable(zs):
+        zs = np.broadcast_to(zs, len(paths))
+
     segs = [_path_to_3d_segment(path, pathz, zdir)
             for path, pathz in zip(paths, zs)]
```

Comment:
That depends a bit on your philosophy, doesn't it? It shouldn't throw an exception since `zip` just stops once one iterable is exhausted, so the plot would just be incomplete. So if your philosphy is to throw exceptions when you think the user is doing something they didn't intend (but legal), then yes. 

I personally would probably leave this up to the user to figure out, but am of course happy to add the check.

Reply:
How would the user figure out something is wrong if it doesn't throw an exception? The previous broadcast _would_ if they didn't match, so we should preserve that.